### PR TITLE
nmstate: Added withAbsent and withVlanInterface functions

### PR DESCRIPTION
**Pull request by evgenLevin**

### CODE REVIEW

File: pkg/nmstate/policy.go [hunk 1]
```diff
@@ -264,6 +264,73 @@ func (builder *PolicyBuilder) WithBondInterface(slavePorts []string, bondName, m
...
+		builder.errorMsg = err.Error()
+
+		return builder
+	}
+
+	glog.V(100).Infof("Creating NodeNetworkConfigurationPolicy %s with VLAN interface configuration: %s %d",
+		builder.Definition.Name, baseInterface, vlanID)
+
+	if baseInterface == "" {
+		glog.V(100).Infof("The baseInterface  can not be empty string")
```
Comment (kononovn) [inline pkg/nmstate/policy.go:279]: Please remove double space

**Review by kononovn — CHANGES_REQUESTED** [review]
just one small comment to fix

File: pkg/nmstate/policy.go (hunk 1 above)
Comment (evgenLevin) [reply pkg/nmstate/policy.go]: fixed

File: pkg/nmstate/policy.go [hunk 2]
```diff
@@ -264,6 +264,73 @@ func (builder *PolicyBuilder) WithBondInterface(slavePorts []string, bondName, m
...
 
+// WithVlanInterface adds VLAN interface configuration to the NodeNetworkConfigurationPolicy.
+func (builder *PolicyBuilder) WithVlanInterface(baseInterface string, vlanID uint16) *PolicyBuilder {
+	if valid, err := builder.validate(); !valid {
+		builder.errorMsg = err.Error()
+
+		return builder
+	}
+
+	glog.V(100).Infof("Creating NodeNetworkConfigurationPolicy %s with VLAN interface configuration: %s %d",
```
Comment (achuzhoy) [inline pkg/nmstate/policy.go:275]: Do you have an example output of this message. It seems like it should be:

Creating NodeNetworkConfigurationPolicy %s with VLAN interface %s and vlanID %d

File: pkg/nmstate/policy.go [hunk 3]
```diff
@@ -264,6 +264,73 @@ func (builder *PolicyBuilder) WithBondInterface(slavePorts []string, bondName, m
...
+		builder.Definition.Name, baseInterface, vlanID)
+
+	if baseInterface == "" {
+		glog.V(100).Infof("The baseInterface can not be empty string")
+
+		builder.errorMsg = "The baseInterface is empty sting"
+	}
+
+	if vlanID > 4094 {
```
Comment (achuzhoy) [inline pkg/nmstate/policy.go:284]: if vlanID > 4094 ||  vlanID < 0 {

File: pkg/nmstate/policy.go [hunk 4]
```diff
@@ -264,6 +264,73 @@ func (builder *PolicyBuilder) WithBondInterface(slavePorts []string, bondName, m
 	return builder.withInterface(newInterface)
 }
 
+// WithVlanInterface adds VLAN interface configuration to the NodeNetworkConfigurationPolicy.
+func (builder *PolicyBuilder) WithVlanInterface(baseInterface string, vlanID uint16) *PolicyBuilder {
+	if valid, err := builder.validate(); !valid {
+		builder.errorMsg = err.Error()
```
Comment (trewest) [inline pkg/nmstate/policy.go:270]: No need to do this, the validate method does it automatically

File: pkg/nmstate/policy.go [hunk 5]
```diff
@@ -264,6 +264,73 @@ func (builder *PolicyBuilder) WithBondInterface(slavePorts []string, bondName, m
...
+		},
+	}
+
+	return builder.withInterface(newInterface)
+}
+
+// WithAbsentInterface appends the configuration for an absent interface to the NodeNetworkConfigurationPolicy.
+func (builder *PolicyBuilder) WithAbsentInterface(interfaceName string) *PolicyBuilder {
+	if valid, err := builder.validate(); !valid {
+		builder.errorMsg = err.Error()
```
Comment (trewest) [inline pkg/nmstate/policy.go:308]: No need to do this, the validate method does it automatically

File: pkg/nmstate/policy.go [hunk 6]
```diff
@@ -264,6 +264,73 @@ func (builder *PolicyBuilder) WithBondInterface(slavePorts []string, bondName, m
...
+		return builder
+	}
+
+	glog.V(100).Infof("Creating NodeNetworkConfigurationPolicy %s with absent interface configuration:"+
+		" interface %s", builder.Definition.Name, interfaceName)
+
+	if interfaceName == "" {
+		glog.V(100).Infof("The interfaceName can not be empty string")
+
+		builder.errorMsg = "The interfaceName is empty sting"
```
Comment (trewest) [inline pkg/nmstate/policy.go:319]:
```suggestion
		builder.errorMsg = "nodenetworkconfigurationpolicy 'interfaceName' cannot be empty"
```

File: pkg/nmstate/policy.go [hunk 7]
```diff
@@ -264,6 +264,73 @@ func (builder *PolicyBuilder) WithBondInterface(slavePorts []string, bondName, m
...
+		return builder
+	}
+
+	glog.V(100).Infof("Creating NodeNetworkConfigurationPolicy %s with VLAN interface configuration: %s %d",
+		builder.Definition.Name, baseInterface, vlanID)
+
+	if baseInterface == "" {
+		glog.V(100).Infof("The baseInterface can not be empty string")
+
+		builder.errorMsg = "The baseInterface is empty sting"
```
Comment (trewest) [inline pkg/nmstate/policy.go:281]:
```suggestion
		builder.errorMsg = "nodenetworkconfigurationpolicy 'baseInterface' cannot be empty"
```

**Review by trewest — CHANGES_REQUESTED** [review]
Just a few changes

File: pkg/nmstate/policy.go (hunk 3 above)
Comment (trewest) [reply pkg/nmstate/policy.go:284]: uint16 cannot be less than 0

File: pkg/nmstate/policy.go (hunk 2 above)
Comment (evgenLevin) [reply pkg/nmstate/policy.go]: changed

File: pkg/nmstate/policy.go (hunk 4 above)
Comment (evgenLevin) [reply pkg/nmstate/policy.go]: removed

File: pkg/nmstate/policy.go (hunk 5 above)
Comment (evgenLevin) [reply pkg/nmstate/policy.go]: removed

File: pkg/nmstate/policy.go (hunk 6 above)
Comment (evgenLevin) [reply pkg/nmstate/policy.go]: removed

File: pkg/nmstate/policy.go (hunk 7 above)
Comment (evgenLevin) [reply pkg/nmstate/policy.go]: removed

**Comment by evgenLevin** [comment]
@trewest @achuzhoy Many thanks for your reviews!  PTAL one more time

**Review by kononovn — APPROVED** [review]
lgtm

**Review by trewest — APPROVED** [review]
/lgtm

**Pull request merged**